presto-hive/src/main/java/io/prestosql/plugin/hive/BackgroundHiveSplitLoader.java

```diff
@@ -553,15 +556,27 @@ private void invokeNoMoreSplitsIfNecessary()
             InternalHiveSplitFactory splitFactory,
             boolean splittable)
     {
-        // Strategy: List once, filter file listing based on manifest file. Complication is that not all files in the manifest are guaranteed
-        // to have a common root.
+        // Strategy: List once, filter file listing based on manifest file.
 
         List<Iterator<LocatedFileStatus>> fileStatusIterators = new ArrayList<>();
         List<Path> paths = new ArrayList<>();
 
         try (BufferedReader reader = new BufferedReader(new InputStreamReader(fileSystem.open(symlinkFile.getPath()), StandardCharsets.UTF_8))) {
             CharStreams.readLines(reader).stream()
-                    .map(Path::new)
+                    .map(line -> {
+                        Path path = new Path(line);
+                        if (path.toUri().getScheme().equals("hdfs")) {
+                            try {
+                                FileSystem hdfs = hdfsEnvironment.getFileSystem(hdfsContext, path);
```

Comment:
path already has uri. Why FS is needed?

Reply:
The authority can be specified multiple ways due to hdfs defaults. E.g the symlink file might contain
hdfs:/foo/bar
hdfs://hadoop-master/foo/bar2
hdfs://hadoop-master:9000/foo/bar3

The listing returns URIs with the authority formatted as in `hdfs.getUri().getAuthority()`. This canonicalization ensures that they match when compared in `.filter(locatedFileStatus -> paths.contains(locatedFileStatus.getPath()))`